src/states/node.rs

```diff
@@ -902,11 +921,12 @@ impl Node {
                 } = hop_msg;
                 self.handle_signed_message(content, route, hop_name, &sent_to)
             }
-            Err(RoutingError::ExceedsRateLimit) => {
+            Err(RoutingError::ExceedsRateLimit(hash)) => {
```

Comment:
Yeah, I was kinda trying to avoid destructuring the message again that way, since I'd want to put `unreachable()` or similar in the `else` block, but that's not great.  I don't think it's too ugly to be passing it back up via the error variant, but I do take your point.  I'd rather leave it as is, but if others agree with you I can make that change easily enough.